pytorch_toolkit/formula_recognition/im2latex/models/text_recognition_heads/attention_based.py

```diff
@@ -93,6 +100,14 @@ def forward(self, features, formulas=None):
         logits: [B, MAX_LEN, VOCAB_SIZE]
         """
         # encoding
+        if self.positional_encodings:
+            old_shape = features.shape
+            pe = PositionalEncodingPermute2D(channels=self.encoder_input_size)
```

Comment:
Is it possible to move it to constructor? Keeping it here, can introduce some extra significant workload for every forward pass.

Reply:
It is made for compatibility with old models, otherwise I should modify `state_dict` of the model in order not to get exception about conflicting keys

Reply:
Why is this? 
You can still keep if conditional on the `positional_encodings` flag. More than that, if `PositionalEncodingPermute2D` is a `nn.Module` with learnable parameters, then current approach is incorrect, because `TextRecognitionHead` is not capable of device management anymore.

Reply:
Moved PE to class constructor